language/stdlib/modules/LibraAccount.move

```diff
@@ -1121,6 +1181,28 @@ module LibraAccount {
         // Check that the chain ID stored on-chain matches the chain ID specified by the transaction
         assert(ChainId::get() == chain_id, Errors::invalid_argument(PROLOGUE_EBAD_CHAIN_ID));
 
+        // Check that the gas calculation will not overflow
+        assert(
+            (txn_gas_price as u128) * (txn_max_gas_units as u128) <= MAX_U64,
+            Errors::invalid_argument(PROLOGUE_ECANT_PAY_GAS_DEPOSIT),
+        );
+
+        let max_transaction_fee = txn_gas_price * txn_max_gas_units;
+        check_sender<Token>(transaction_sender, max_transaction_fee, txn_sequence_number, txn_public_key);
+
+        assert(
+            LibraTimestamp::now_seconds() < txn_expiration_time_seconds,
+            Errors::invalid_argument(PROLOGUE_ETRANSACTION_EXPIRED)
+        );
+    }
+
+    /// Check if sender is able to send a transaction
+    public fun check_sender<Token>(
```

Comment:
I don't see any reason to introduce this "check_sender" function and rearrange the various "prologue_common" checks as you've done here. It seems like this is just an artifact of the previous version of the code, and if so, I think you should leave out those changes.

Reply:
My thought is that exposing this function would be easier for us to refactor in the future, as it rips off the account related part. We can get rid of this function for sure as well.